src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -712,8 +712,13 @@ void Lowering::LowerSIMD(GenTreeSIMD* simdNode)
                 BlockRange().Remove(list->Current());
             }
 
+            assert(sizeof(constArgValues) == 16);
+
+            UNATIVE_OFFSET cnsSize  = sizeof(constArgValues);
+            UNATIVE_OFFSET cnsAlign = (comp->compCodeOpt() != Compiler::SMALL_CODE) ? cnsSize : 1;
```

Comment:
x86 supports unaligned loads so we can allow unaligned constants when `SMALL_CODE`